### PR TITLE
perf: mediainfo router should use async/await to get file info

### PR DESCRIPTION
**Pull request by trim21**
## Description

current mediainfo will block nodejs process, use async/await to avoid blocking

now we are targeting to node18, so also upgrade `"@types/node"` to `^18`

## Types of changes

- [ ] Breaking change (changes that break backward compatibility of public API or CLI - semver MAJOR)
- [ ] New feature (non-breaking change which adds functionality - semver MINOR)
- [X] Bug fix (non-breaking change which fixes an issue - semver PATCH)

### CODE REVIEW

File: server/services/qBittorrent/clientRequestManager.ts [hunk 1]
```diff
@@ -483,7 +483,7 @@ class ClientRequestManager {
         new URLSearchParams({
           hashes: hashes.join('|').toLowerCase(),
           tags: tags?.join(','),
-        }),
+        } as Record<string, string>),
```
Comment (trim21) [inline server/services/qBittorrent/clientRequestManager.ts:486]: this and changes in `server/services/rTorrent/util/scgiUtil.ts` are caused by `@types/node`

**Comment by jesec** [comment]
Looks good in general. Thanks for the PR.

Some nits:

- Don't bump the versionof Node.js type yet. We still want to maintain compatibility as much as possible. If we do want to bump, make clear what features require it.
- I think we can use `fs.promises.` instead of having a `fsp`

**Comment by trim21** [comment]
> Looks good in general. Thanks for the PR.
> 
> Some nits:
> 
> * Don't bump the versionof Node.js type yet. We still want to maintain compatibility as much as possible. If we do want to bump, make clear what features require it.
> * I think we can use `fs.promises.` instead of having a `fsp`

OK， I revert version of `@types/node` back to 12

File: server/util/async.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export async function asyncFilter<T>(
+  arr: Array<T>,
+  predicate: (item: T, index: number) => Promise<boolean>,
+): Promise<Array<T>> {
+  const results = await Promise.all(arr.map(predicate));
```
Comment (trim21) [inline server/util/async.ts:5]: should use for loop and await to avoid too many promises maybe?

Comment (jesec) [reply server/util/async.ts]: Yeah. That's a good idea.

Comment (trim21) [reply server/util/async.ts]: then it's already done...

**Pull request merged**